### PR TITLE
[#109] Part II: Improve unit tests for Context

### LINKED ISSUE #109: Improve unit tests for `SSHKit` and `Context`
The unit tests for `SSHKit` and `SSHKit.Context` have room for improvement and should be more extensive.

### PR DESCRIPTION
**Pull request by pmeinhardt**
**Fixes #109 (part 2). Adds tests for `SSHKit.Context` functions.**

---

* [ ] Documented the change if necessary
* [x] Tested this PRs change (with unit and/or functional tests)
* [ ] Added this PRs change to CHANGELOG.md (in the `#master` section) if  necessary.

### CODE REVIEW

File: lib/sshkit/context.ex [hunk 1]
```diff
@@ -40,9 +40,9 @@ defmodule SSHKit.Context do
   end
 
   defp sudo(command, nil, nil), do: command
-  defp sudo(command, username, nil), do: "sudo -n -u #{username} -- sh -c #{shellquote(command)}"
-  defp sudo(command, nil, groupname), do: "sudo -n -g #{groupname} -- sh -c #{shellquote(command)}"
-  defp sudo(command, username, groupname), do: "sudo -n -u #{username} -g #{groupname} -- sh -c #{shellquote(command)}"
+  defp sudo(command, username, nil), do: "sudo -H -n -u #{username} -- sh -c #{shellquote(command)}"
+  defp sudo(command, nil, groupname), do: "sudo -H -n -g #{groupname} -- sh -c #{shellquote(command)}"
+  defp sudo(command, username, groupname), do: "sudo -H -n -u #{username} -g #{groupname} -- sh -c #{shellquote(command)}"
```
Comment (pmeinhardt) [inline lib/sshkit/context.ex]: https://explainshell.com/explain?cmd=sudo+-H+-n+-u+me+--+sh+-c+%27%2Fusr%2Fbin%2Fenv+echo+%24HOME%27

File: lib/sshkit/context.ex [hunk 2]
```diff
@@ -28,7 +28,7 @@ defmodule SSHKit.Context do
   "cd /var/www && /usr/bin/env ls"
 
   iex> %SSHKit.Context{user: "me"} |> SSHKit.Context.build("whoami")
-  "sudo -n -u me -- sh -c \\"/usr/bin/env whoami\\""
+  "sudo -H -n -u me -- sh -c '/usr/bin/env whoami'"
```
Comment (klappradla) [inline lib/sshkit/context.ex]: 👍

File: lib/sshkit/context.ex (hunk 1 above)
Comment (klappradla) [reply lib/sshkit/context.ex]: Cool! Did not know about this! So your refactoring of the tests here already found and fixed a potential source for bugs 🐛 ???

**Comment by pmeinhardt** [comment]
Thanks a lot for taking the time to read and understand this @klappradla 👌 
I'm really glad it made things clearer 😃

**Pull request merged**